programs/token/src/state/mint.rs

```diff
@@ -0,0 +1,98 @@
+use crate::ID;
+
+use pinocchio::{
+    account_info::{AccountInfo, Ref},
+    program_error::ProgramError,
+    pubkey::Pubkey,
+};
+pub struct Mint(*const u8);
+
+impl Mint {
+    pub const LEN: usize = 82;
+
+    /// Performs owner and length validation on `AccountInfo` and returns a `Ref<T>` for safe borrowing.
+    pub fn from_account_info(account_info: &AccountInfo) -> Result<Ref<Mint>, ProgramError> {
+        if account_info.data_len() != Self::LEN {
+            return Err(ProgramError::InvalidAccountData);
+        }
+        if account_info.owner() != &ID {
+            return Err(ProgramError::InvalidAccountData);
+        }
+        Ok(Ref::map(account_info.try_borrow_data()?, |data| unsafe {
+            &*(data.as_ptr() as *const Mint)
+        }))
+    }
+
+    /// # Safety
+    /// Performs owner and length validation on `AccountInfo` but performs unchecked borrowing and
+    /// returns a `T` directly.
+    #[inline(always)]
+    pub unsafe fn from_account_info_unchecked(
+        account_info: &AccountInfo,
+    ) -> Result<Mint, ProgramError> {
+        if account_info.data_len() != Self::LEN {
+            return Err(ProgramError::InvalidAccountData);
+        }
+        if account_info.owner() != &ID {
+            return Err(ProgramError::InvalidAccountData);
+        }
+        Ok(Self::from_bytes(account_info.borrow_data_unchecked()))
+    }
+
+    /// # Safety
+    /// Constructs a `T` directly from a byte slice. The caller must ensure that `bytes` contains a
+    /// valid representation of `T`.
+    pub unsafe fn from_bytes(bytes: &[u8]) -> Self {
+        Self(bytes.as_ptr())
+    }
+
+    #[inline(always)]
+    pub fn has_mint_authority(&self) -> bool {
+        unsafe { *(self.0 as *const bool) }
```

Comment:
hopefully whatever is storing this isn't storing bool and is storing u8, and you should also load as u8 and explicitly check for != 0